src/sortStudents.ts

```diff
@@ -1,16 +1,91 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copy = [...students];
+  const callback = (acc: number, current: number): number => {
+    return acc + current;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        return copy.sort((a, b) => {
+          return a.name.localeCompare(b.name);
+        });
+      }
+
+      return copy.sort((a, b) => {
+        return b.name.localeCompare(a.name);
+      });
+
+    case SortType.Surname:
+      if (order === 'asc') {
+        return copy.sort((a, b) => {
+          return a.surname.localeCompare(b.surname);
+        });
+      }
+
+      return copy.sort((a, b) => {
+        return b.surname.localeCompare(a.surname);
+      });
```

Comment:
you can group similar cases like this.
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
```